src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Reliability/Common/SqlRetryIntervalBaseEnumerator.cs

```diff
@@ -0,0 +1,110 @@
+
+        /// <include file='../../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlRetryIntervalBaseEnumerator.xml' path='docs/members[@name="SqlRetryIntervalBaseEnumerator"]/MinTimeInterval/*' />
+        public TimeSpan MinTimeInterval { get; protected set; }
+
+        /// <include file='../../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlRetryIntervalBaseEnumerator.xml' path='docs/members[@name="SqlRetryIntervalBaseEnumerator"]/Current/*' />
+        public TimeSpan Current { get; protected set; } = TimeSpan.Zero;
+
+        object IEnumerator.Current => Current;
+
+        /// <include file='../../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlRetryIntervalBaseEnumerator.xml' path='docs/members[@name="SqlRetryIntervalBaseEnumerator"]/ctor1/*' />
+        public SqlRetryIntervalBaseEnumerator()
+        {
+            GapTimeInterval = TimeSpan.Zero;
+            MaxTimeInterval = TimeSpan.Zero;
+            MinTimeInterval = TimeSpan.Zero;
+        }
+
+        /// <include file='../../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlRetryIntervalBaseEnumerator.xml' path='docs/members[@name="SqlRetryIntervalBaseEnumerator"]/ctor2/*' />
+        public SqlRetryIntervalBaseEnumerator(TimeSpan timeInterval, TimeSpan maxTime, TimeSpan minTime)
+        {
+            Validate(timeInterval, maxTime, minTime);
+            GapTimeInterval = timeInterval;
+            MaxTimeInterval = maxTime;
+            MinTimeInterval = minTime;
+        }
+
+        /// <include file='../../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlRetryIntervalBaseEnumerator.xml' path='docs/members[@name="SqlRetryIntervalBaseEnumerator"]/Reset/*' />
+        public virtual void Reset()
+        {
+            Current = TimeSpan.Zero;
+        }
+
+        /// <include file='../../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlRetryIntervalBaseEnumerator.xml' path='docs/members[@name="SqlRetryIntervalBaseEnumerator"]/Validate/*' />
+        protected virtual void Validate(TimeSpan timeInterval, TimeSpan maxTimeInterval, TimeSpan minTimeInterval)
+        {
+            if(minTimeInterval < _minValue || minTimeInterval > _maxValue )
+            {
+                throw SqlCRLUtil.ArgumentOutOfRange(nameof(minTimeInterval), minTimeInterval, _minValue, _maxValue);
+            }
+
+            if (maxTimeInterval < _minValue || maxTimeInterval > _maxValue)
+            {
+                throw SqlCRLUtil.ArgumentOutOfRange(nameof(maxTimeInterval), maxTimeInterval, _minValue, _maxValue);
+            }
+
+            if (timeInterval < _minValue || timeInterval > _maxValue)
+            {
+                throw SqlCRLUtil.ArgumentOutOfRange(nameof(timeInterval), timeInterval, _minValue, _maxValue);
+            }
+
+            if (maxTimeInterval < minTimeInterval)
+            {
+                throw SqlCRLUtil.ArgumentOutOfRange(nameof(maxTimeInterval), maxTimeInterval, minTimeInterval, _maxValue);
```

Comment:
If this condition is true, it doesn't necessarily mean `maxTimeInterval` is the invalid value.